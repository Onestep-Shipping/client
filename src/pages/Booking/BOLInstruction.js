import React, {useState, useCallback} from 'react';
import './Booking.css';

import {useParams, useHistory} from 'react-router-dom';
import DATA from '../../data/ScheduleDetailsData.js';
import Header from '../../components/Header/Header.js';

const ROW_NUMBER = 5;

const BOLInstruction = () => {
  const {id} = useParams();
  const history = useHistory();

  const TRACKING_HEADERS = ["Container No.", "Seel No.", "Cargo Weight (kgs)", "Measurement (cbm)", "VGM"];
  const [row, setRow] = useState(1);

  const handleBook = useCallback((e) => {
    e.preventDefault();
    // const shipper = e.target[0].value;
    // const consignee = e.target[1].value;
    // const notify = e.target[2].value;
    // const orderNo = e.target[3].value;
    // const description = e.target[4].value;
    // const hsCode = e.target[5].value;
    // const caedNo = e.target[6].value;
    // history.push(id + '/completed');
  }, [history, id]);

  const addContainer = () => {
    setRow(row + 1);
  }

  return (
    <div className="homepage-container">
      <Header />
      <div className="body-container2">
        <div className="booking-container">
          <div className="h1-container">
            <h1>Bill of Lading Instruction - Booking #212314</h1>
          </div>
          <div className="info-container2">
            {DATA.map((booking, index) => {
              if (index.toString() === id) {
                return (
                  <div className="booking-details-container2" key={index}>
                    <div className="info-row">
                      <div>
                        <text className="info-label">From: </text>
                        <text className="schedule-info-text-left">{booking.from}</text>
                      </div>
                      <div>
                        <text className="info-label">Date: </text>
                        <text className="schedule-info-text">{booking.fromDate}</text>
                      </div>
                    </div>
                    <div className="info-row">
                      <div>
                        <text className="info-label">To: </text>
                        <text className="schedule-info-text-left">{booking.to}</text>
                      </div>
                      <div>
                        <text className="info-label">Date: </text>
                        <text className="schedule-info-text">{booking.toDate}</text>
                      </div>
                    </div>
                    <div className="info-row">
                      <text className="info-label">Vessels / Services: </text>
                      <text className="schedule-result-text">{booking.ves}</text>
                    </div>
                  </div>
                );
              } else {
                return null;
              }
            })}
            <form className="booking-form-container" onSubmit={handleBook} noValidate>
              <div className="info-row-with-textarea">
                <text className="info-label">Shipper: </text>
                <textarea
                  type="text"
                  name="shipper"
                  className="booking-form-textarea"
                  placeholder="i.e. Food, Clothes, etc."
                  rows={ROW_NUMBER}
                  required/>
              </div>
              <div className="info-row-with-textarea">
                <text className="info-label">Consignee: </text>
                <textarea
                  type="text"
                  name="consignee"
                  className="booking-form-textarea"
                  placeholder="i.e. Food, Clothes, etc."
                  rows={ROW_NUMBER}
                  required/>
              </div>
              <div className="info-row-with-textarea">
                <text className="info-label">Notify: </text>
                <textarea
                  type="text"
                  name="notify"
                  className="booking-form-textarea"
                  placeholder="i.e. Food, Clothes, etc."
                  rows={ROW_NUMBER}
                  required/>
              </div>
              <div className="info-row-with-textarea">
                <text className="info-label">Description of Goods: </text>
                <textarea
                  type="text"
                  name="notify"
                  className="booking-form-textarea"
                  placeholder="i.e. Food, Clothes, etc."
                  rows={ROW_NUMBER}
                  required/>
              </div>
              <div className="schedule-result-header-row">
              {
                TRACKING_HEADERS.map((header, ind) => {
                  return (
                    <div className="col2" key={ind}>
                      <text className="schedule-result-header-text">{header}</text>
                    </div>
                  );
                })
              }
              </div>
              {Array(row).fill().map((booking, ind) => {
                return (
                  <div className="schedule-result-header-row" key={ind}>
                    <div className='instruction-result-row'>
                      <div className="col2">
                        <input
                          type="text"
                          name="order-number"
                          className="instruction-extra-input"
                          placeholder="i.e. 420010"
                          required/>
                      </div>
                      <div className="col2">
                        <input
                          type="number"
                          name="order-number"
                          className="instruction-extra-input"
                          placeholder="i.e. 420010"
                          required/>
                      </div>
                      <div className="col2">
                        <input
                          type="number"
                          name="order-number"
                          className="instruction-extra-input"
                          placeholder="i.e. 420010"
                          required/>
                      </div>
                      <div className="col2">
                        <input
                          type="number"
                          name="order-number"
                          className="instruction-extra-input"
                          placeholder="i.e. 420010"
                          required/>
                      </div>
                      <div className="col2">
                        <input
                          type="number"
                          name="order-number"
                          className="instruction-extra-input"
                          placeholder="i.e. 420010"
                          required/>
                      </div>
                    </div>
                  </div>
                );
              })}

              <button className="add-button" onClick={addContainer}>
                Add Container
              </button>
              
              <div className="info-row">
                <text className="info-label">Order/PO Number: </text>
                <input
                  type="text"
                  name="order-number"
                  className="booking-form-input"
                  placeholder="i.e. 420010"
                  required/>
              </div>
              <div className="info-row">
                <text className="info-label">HS Code: </text>
                <input
                  type="number"
                  name="hs-code"
                  className="booking-form-input"
                  placeholder="i.e. 420010"
                  required/>
              </div>
              <div className="info-row">
                <text className="info-label">CAED/AES Number: </text>
                <input
                  type="text"
                  name="caed-number"
                  className="booking-form-input"
                  placeholder="i.e. 420010"
                  required/>
              </div>
              <div className="info-row">
                <text className="info-label">Cargo Value: </text>
                <input
                  type="text"
                  name="cargo-value"
                  className="booking-form-input"
                  placeholder="i.e. 420010"
                  required/>
              </div>
              <button className="result-button">
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default BOLInstruction;
