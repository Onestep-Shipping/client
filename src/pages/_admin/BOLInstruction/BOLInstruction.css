.bol-instruction-container {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.bol-instruction-list {
  width: 30%;
  height: 100%;
  list-style-type: none;
  background-color: white;
  margin: 0;
  padding: 0;
  overflow:hidden; 
  overflow-y:scroll;
}

.bol-instruction-detail {
  height: 100%;
  width: 70%;
  background-color: pink;
}

.bol-instruction-item {
  width: 100%;
  height: 100px;
  background-color: aliceblue;
  border: 2px solid white;
}