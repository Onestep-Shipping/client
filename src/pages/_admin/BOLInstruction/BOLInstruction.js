import React, {useState, useRef} from 'react';
import './BOLInstruction.css';

import Header from '../../../components/Header/Header.js';
import BookingDisplay from '../../../components/BookingDisplay/BookingDisplay.js';
import UserList from '../../../components/UserList/UserList.js';
import BOL from '../../../data/BOLInstructionData.js';
import {InfoRow, ShipmentDetail} from '../Helpers.js';
import FileUploadService from '../../../services/FileUploadService.js';

const BOLInstruction = () => {
  const [currentBolIndex, setCurrentBolIndex] = useState(0);
  const [pdf, setPDF] = useState('')
  const inputFile = useRef(null) 

  const handleChange = () => {
    const pathSplit = document.getElementById("file").value.split("\\");
    const newVal = pathSplit[pathSplit.length - 1]
    if (newVal !== "") {
      setPDF(newVal);
    }
  }

  const handleCurrentBolIndexChange = ind => {
    if (ind !== currentBolIndex) {
      setPDF("");
    }
    setCurrentBolIndex(ind);
  }

  const handleSubmit = e => {
    e.preventDefault();
    const formData = new FormData(e.currentTarget);
    FileUploadService.uploadFile(formData)
      .then(res => {
        console.log(res.data);
      })
      .catch(e => {
        console.log(e);
      });
  }

  return (
    <div className="homepage-container">
      <Header />
      <div className="bol-instruction-container">
        <UserList 
          setInd={handleCurrentBolIndexChange}
          opt={BOL} type="bol" />
        <div className="bol-instruction-detail"> 
          <div className="booking-id-container">
            <h1>BOL Instruction</h1>
          </div>
          <div className="customer-info-container">
            <text>Email: {BOL[currentBolIndex].email}</text>
            <text>{BOL[currentBolIndex].dateSent}</text>
          </div>

          <div className="form-container">
            <h2>Schedule</h2>
            <BookingDisplay id={'' + currentBolIndex} fields={3}/>
          </div>

          <div className="form-container">
            <h2>BOL Instruction</h2>
            <div className="booking-details-container">
              <InfoRow label="Shipper" value={BOL[currentBolIndex].bol.shipper} />
              <InfoRow label="Consignee" value={BOL[currentBolIndex].bol.consignee} />
              <InfoRow label="Notify" value={BOL[currentBolIndex].bol.notify} />
              <InfoRow label="Description of Goods" value={BOL[currentBolIndex].bol.description} />
              <InfoRow label="Shipment Detail" value="" />
              <div className="shipment-detail-row">
                {BOL[currentBolIndex].bol.cargoInfo.map((container, ind) => (
                 <ShipmentDetail key={ind} container={container} ind={ind} />
                ))}
              </div>
              <InfoRow label="Order/PO Number" value={BOL[currentBolIndex].bol.orderNo} />
              <InfoRow label="HS Code" value={BOL[currentBolIndex].bol.hsCode} />
              <InfoRow label="CAED/AES Number" value={BOL[currentBolIndex].bol.caedNo} />
              <InfoRow label="Cargo Value" value={BOL[currentBolIndex].bol.value} />
            </div>
          </div>

          <div className="pdf-text">{pdf}</div>
          {!BOL[currentBolIndex].isCompleted &&
            <form 
              className="bol-button-form" 
              encType="multipart/form-data"
              onSubmit={handleSubmit}>
              <label htmlFor="file" className="file-label"> 
                Choose a file
              </label>
              <input 
                name="file"
                type='file' id='file' ref={inputFile} 
                className="inputfile" 
                onChange={handleChange}
                accept="application/pdf"
              />
              <input type="submit" className="result-button" value="Send to Customer" />
            </form>
          }
        </div>
      </div>
    </div>
  );
};

export default BOLInstruction;
