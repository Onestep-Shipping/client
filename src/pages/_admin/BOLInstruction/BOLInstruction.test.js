import { ApolloProvider } from '@apollo/react-hooks';
import BOLInstruction from './BOLInstruction';
import React from 'react';
import { shallow } from 'enzyme';

let wrapper = shallow(
  <ApolloProvider>
    <BOLInstruction />
  </ApolloProvider>
);

describe('BOLInstruction', () => {
  it('should render the BOLInstruction Component correctly', () => {   
    expect(BOLInstruction).toMatchSnapshot();
  });
});