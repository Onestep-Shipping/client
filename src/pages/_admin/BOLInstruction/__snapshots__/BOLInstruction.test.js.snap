// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BOLInstruction should render the BOLInstruction Component correctly 1`] = `[Function]`;
